### DISCUSSION TITLE
Using load: directoryLoader

### DISCUSSION THREAD

**Discussion by RavenPoe1900** [Nikaple/nest-typed-config#207] [Q&A] [answered]
Hi, when i set load: directory Loader i get the following error.

[ExceptionHandler] Nest can't resolve dependencies of the AppService (RootConfig, ?, TableConfig). Please make sure that the argument DatabaseConfig at index [1] is available in the AppModule context.

Potential solutions:
- If DatabaseConfig is a provider, is it part of the current AppModule?
- If DatabaseConfig is exported from a separate @Module, is that module imported within AppModule?
  @Module({
    imports: [ /* the Module containing DatabaseConfig */ ]
  })

My code is the example. I create a new route for my environment configuration with the following configuration:

> `@Module({`
>   `imports: [`
>     `TypedConfigModule.forRoot({`
>      ` schema: RootConfig,`
>       `// load: fileLoader(),`
>      ` load: directoryLoader({`
>        ` directory: `****/config/env`,`
>       `//   /* other cosmiconfig options */
`>       `}),`
>     `}),`
>   `],`
>   `providers: [AppService],`
>   `controllers: [AppController],`
> `})`
> `export class AppModule {}`

and 

### // app.service.ts
import { Injectable } from '@nestjs/common';
import { DatabaseConfig, RootConfig, TableConfig} from './config';

@Injectable()
export class AppService {
  // inject any config or sub-config you like
  constructor(
    private readonly config: RootConfig,
    private readonly databaseConfig: DatabaseConfig,
    private readonly tableConfig: TableConfig,
  ) {}

  // enjoy type safety!
  public show(): string {
    const out = [
      `root.name: ${this.config.database.table.name}`,
      `root.database.name: ${this.databaseConfig.table.name}`,
      `root.database.table.name: ${this.tableConfig.name}`,
    ].join('\n');

    return `${out}\n`;
  }
}

**Comment by Nikaple** [comment]
Please provide a [minimum reproduction repo](https://minimum-reproduction.wtf) to help better investigation.

**Reply by Nikaple** [reply]
Or you can play around with existing CodeSandboxes in the README, and compare it to what you've got so far.

**Reply by RavenPoe1900** [reply]
This is the link of my project https://github.com/RavenPoe1900/nest-typed-config

**Comment by Nikaple** [comment, marked as answer]
Directory loader is used when you want to split config into small files, and each file name represents a namespace. For example, `foo.json` with `{"bar": 1}` will become `{"foo":{"bar":1}}`. You should use `fileLoader` instead.
